EDITING.md

```diff
@@ -0,0 +1,177 @@
+
+If you've considered all these things, and still think that you've got a good
+change in mind, then thank you for the depth of your thoughts so far!
+Go ahead and either create an issue on github, or hoist a PR, or move forward
+by writing up an [exploration report](#use-exploration-reports) to help
+hone further discussion, or, simply find us and start a conversation on
+the `#ipld` channel on freenode IRC!
+
+
+Use Exploration Reports
+-----------------------
+
+One way to make progress in designing something or discussing a tricky issue
+is to make an "exploration report".
+
+The key point of an "exploration report" is to document some thinking
+and collate notes about a complex topic, *without* jumping straight to a
+proposed solution.  This is useful because we can readily preserve the notes,
+even if the idea in question doesn't pan out or can't be completed in one joust.
+
+We've got a bunch of these -- and more description of what that concept is --
+in the [`./design/history/exploration-reports`](./design/history/exploration-reports)
+directory.  You can also make exploration reports as gists, or other kinds of
+documents, or even issues in github; whatever creates the least burden is best.
+
+Exploration reports are almost guaranteed to be a good idea if proposing
+any kind of [semantic change](#semantic-changes).
+
+
+Stylistic Preferences
+---------------------
+
+### headings
+
+Be careful with headings -- they should look good as links.
+Being able to link deeply into a spec document is every bit as important
+as being able to read it from top to bottom.
+
+Be very careful not to have headings that are repeated; this makes linking to that heading impossible.
+(If you feel that repeated headings are still visually good formating,
+consider if a `"**bold text**: subject details"` approach fits.
+Because this doesn't try to produce links, it's not a problem to repeat the bold component.
+But, by the other side of the same coin, _it can't be linked to_; be mindful of the concession.)
+
+Don't put redundant information or parenthetical remarks in a heading;
+it rarely comes out legible in the resulting links.
+For example, "#block-layer" is preferable to "#block-layer-layer-0".
+Clarifications and secondary terminology variations are better expressed in the body text
+than jammed into the heading and adding tongue-twisters to links.
+
+### linking
+
+The preferred linking format depends on what you're linking to and how far away it is.
+In general, the aim is to make editing easier in the long run.
+
+- The preferred link syntax depends on how distant the target is.
+  - Links within a document should use bare hashtags -- `[text](#heading)`.
+  - Links to documents in the same folder should use relative paths -- `[text](./file.md#heading)`.
+  - Links to documents elsewhere in the repo should use rooted paths -- `[text](/full/path.md#heading)`.
```

Comment:
This comment is kind void after I was finished writing it and thinking about things. I post it though for future reference.

I usually prefer having relative links as they don't break when you serve things up as HTML where the docs are not at the root. This is a tooling issue, but sadly so many tools are broken in this regards.

But I like your reasoning (easier to search) and so for me the question is now "how often do I serve it up and how often do I just read the plain markdown". And the answer is clearly that I read the plain Markdown or rendered on GitHub where absolute links work.